Neos.Neos/Classes/Domain/NodeLabel/ExpressionBasedNodeLabelGenerator.php

```diff
@@ -63,9 +61,37 @@ public function setExpression(string $expression): void
      */
     public function getLabel(Node $node): string
     {
-        if (Utility::parseEelExpression($this->getExpression()) === null) {
-            return $this->getExpression();
+        $contentRepository = $this->contentRepositoryRegistry->get($node->contentRepositoryId);
+        $nodeTypeManager = $contentRepository->getNodeTypeManager();
+        $nodeTypeManager->getNodeType($node->nodeTypeName);
+        $subgraph = $this->contentRepositoryRegistry->subgraphForNode($node);
+
+        $expression = $this->getExpression();
+        $parentNode = $subgraph->findParentNode($node->aggregateId);
+        $parentNodeType = $parentNode ? $nodeTypeManager->getNodeType($parentNode->nodeTypeName) : null;
+
+        if ($parentNodeType && $this->isAutoCreatedChildNode(
```

Comment:
before fetching the `$parentNode` id like to make a non expensive check if the `$node->classification->isTethered()` because otherwise we can just not do that :D

Reply:
Its kindof unfortunate that fetching the label is now an expensive operation also for main content collections etc. but this is the pain for crazy comfort :)

Reply:
Is the check `$nodeTypeOfParent->tetheredNodeTypeDefinitions->contain($nodeName)` still needed afterwards?

Reply:
nope not really because by accessing `.childNodes` like you do safely this is the same